### ISSUE TITLE
bug: node.js socket 'data' events not firing in electron plugin

### ISSUE THREAD

**Issue by Alphapork** [ionic-team/capacitor#2788]
# Bug Report

## Capacitor Version

`npx cap doctor` output:

   Capacitor Doctor   

Latest Dependencies:

  @capacitor/cli: 2.0.1

  @capacitor/core: 2.0.1

  @capacitor/android: 2.0.1

  @capacitor/ios: 2.0.1

Installed Dependencies:

  @capacitor/android not installed

  @capacitor/ios not installed

  @capacitor/cli 1.5.1

  @capacitor/core 1.5.1

[fatal] Platform electron is not valid.

## Affected Platform(s)

- [ ] Android
- [ ] iOS
- [x] Electron
- [ ] Web

## Current Behavior

After creating, connecting and regestering an event listener for the 'data' event on a socket inside an electron plugin the event isn't firing when recieving data. The application connects, and sometimes it does register a couple of events before going silent, but in most cases it is completely silent after connecting.

## Expected Behavior

'data' events should be firing when the socket recieves data.

## Sample Code or Sample Application Repo
```
import { WebPlugin } from '@capacitor/core';
import { ElectronSocketsPlugin } from './definitions';

const net = window.require('net');

//let socket: any = null;
export class ElectronSocketsWeb extends WebPlugin implements ElectronSocketsPlugin {
  socket: any = null;
  constructor() {
    super({
      name: 'ElectronSockets',
      platforms: ['electron']
    });

  }

  async createSocket(recvFunc: (data: any) => void) {
    this.socket = net.Socket();
    let thisRef = this;
    this.socket.connect(5001, 'localhost', function () {
      console.log('Connected');
    });
    this.socket.on('data', recvFunc);
  }

  async echo(options: { value: string }): Promise<{ value: string }> {
    console.log('ECHO', options);
    return options;
  }
}

const ElectronSockets = new ElectronSocketsWeb();

export { ElectronSockets };

import { registerWebPlugin } from '@capacitor/core';
registerWebPlugin(ElectronSockets);
```

## Reproduction Steps

1. Import the plugin described above into an application and call createSocket(recvFunc) with a callback function that logs that a chunk has been received. 

2. create a netcat server listening on port 5001.

3. Start application. When the connection has been established, send data from the server and check if the event fires.

## Other Technical Details

`npm --version` output: 6.13.4

`node --version` output: v12.16.1

`pod --version` output (iOS issues only):

## Other Information

I have checked the other events on the socket as well, but nothing fires after the 'connected' event.
I don't know why npx doctor gives me the output `[fatal] Platform electron is not valid.`, but I can build and run as an electron app.

**Comment by jcesarmobile** [comment]
Sorry, this doesn't look like a Capacitor bug, more like the plugin is not correctly build.

The `Platform electron is not valid` was fixed in Capacitor 2, you are still on 1.5.1.

**Comment by Alphapork** [comment]
That might be true. I have tried to find documentation on how to correctly create an electron plugin but havn't had any luck. Maybe you can point me in the right direction?
Should what I'm trying to do be possible?

**Comment by Alphapork** [comment]
> Sorry, this doesn't look like a Capacitor bug, more like the plugin is not correctly build.
> 
> The `Platform electron is not valid` was fixed in Capacitor 2, you are still on 1.5.1.

That might be true. I have tried to find documentation on how to correctly create an electron plugin but havn't had any luck. Maybe you can point me in the right direction?
Should what I'm trying to do be possible?
